### ISSUE TITLE
`@stylistic/type-generic-spacing`: Broken on interface definition

### ISSUE THREAD

**Issue by ej612** [eslint-stylistic/eslint-stylistic#438]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The rule reports a problem on the following code:
```ts
export interface MyType<Type extends (boolean | string) = (boolean | string)> {
    id: string,
    value: Type
}
```

![image](https://github.com/eslint-stylistic/eslint-stylistic/assets/110408767/252290ff-1d6e-4555-9fee-316df39474c6)

It auto-fixes to the following, which is broken:

```ts
export interface MyType<Type extends (boolean | string = boolean | string)> {
    id: string,
    value: Type
}
```

![image](https://github.com/eslint-stylistic/eslint-stylistic/assets/110408767/ceedc74e-9226-480f-ac01-5af0a16fe1c2)

### Reproduction

N/A

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
